packages/babel-parser/src/parser/statement.ts

```diff
@@ -314,6 +314,17 @@ export default abstract class StatementParser extends ExpressionParser {
     return false;
   }
 
+  startsUsingForOf(): boolean {
+    const lookahead = this.lookahead();
+    if (lookahead.type === tt._of && lookahead.end - lookahead.start === 2) {
```

Comment:
Can you add a comment saying that we check that the length is 2 because of escape sequences?

Reply:
I end up using the `containsEsc` flag.